### PR TITLE
Automatically open an issue for every new release

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Automatically creates issues for every (new) release. For an example see https://github.com/dblock/opensearch-build/issues/20. It will leave existing issues that match the title alone, so if you close an issue it won't reopen a new one or edit any body/title. 

The latter required some code changes in https://github.com/JasonEtco/create-an-issue (https://github.com/JasonEtco/create-an-issue/pull/112 and https://github.com/JasonEtco/create-an-issue/pull/111). I made [a release](https://github.com/dblock/create-an-issue/releases/tag/v2.5.1-opensearch) in my fork with these changes.

With every new release we can keep improving the checklist!
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
@peternied what else should go into the checklist?

**Comment by dblock** [comment]
@peternied @mch2 @saratvemulapalli This is ready.

File: .github/ISSUE_TEMPLATE/release_template.md
Comment (saratvemulapalli) [inline .github/ISSUE_TEMPLATE/release_template.md:35]: Add:
- Cut release tags in each component.

I would take it further. Let know what you think. 
Use `.github meta` to create issues in each component.
Probably it would look like 
- [Meta] Issue for Preparation in `opensearch-build`, create individual issues in each component via `meta` with the checklist. 
- Similarly for each subsection of this release template.
- Link all the meta issues to this main release issue.

Comment (dblock) [reply .github/ISSUE_TEMPLATE/release_template.md:36]: Updated. I mentioned meta and such.

I think creating an issue in each component repo can also get automated after this PR is merged. As you correctly point out, each component needs to have a checklist of creating a branch, adding themselves to the manifest, etc. 

We don't have to get this perfect rn and can keep iterating on the template as we go.

Comment (saratvemulapalli) [reply .github/ISSUE_TEMPLATE/release_template.md]: Absolutely. Just putting in food for thought.

**Review by saratvemulapalli — APPROVED** [review]
Nice to see these automations rolling :)

File: .github/ISSUE_TEMPLATE/release_template.md
Comment (peternied) [inline .github/ISSUE_TEMPLATE/release_template.md:26]: In this section or as part of the release section

---

Verify all {{ env.VERSION }} labeled issues have been triaged and resolved.

**Review by peternied — DISMISSED** [review]
Minor comment

**Review by mch2 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/release_template.md
Comment (dblock) [reply .github/ISSUE_TEMPLATE/release_template.md:26]: adding

**Comment by dblock** [comment]
Gah, an update dismissed my review :(

**Review by saratvemulapalli — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Pull request merged**
